src/models/tools/geometry/jxg-movable-line.ts

```diff
@@ -10,6 +10,41 @@ export const isMovableLine = (v: any) => {
 
 export const isVisibleMovableLine = (v: any) => isMovableLine(v) && v.visProp.visible;
 
+export const isMovableLineEquation = (v: any) => {
+  return v instanceof JXG.Text && v.getAttribute("clientType") === kMovableLineType;
+};
+
+// Returns the two points where the given line intersects the given board, sorted from left to right
+export const getBoundingBoxIntersections = (slope: number, intercept: number, board: JXG.Board) => {
+  const boundingBox = board.attr.boundingbox;
+  const leftX = boundingBox[0];
+  const topY = boundingBox[1];
+  const rightX = boundingBox[2];
+  const bottomY = boundingBox[3];
+  const topIntersection = [solveForX(slope, intercept, topY), topY];
+  const rightIntersection = [rightX, solveForY(slope, intercept, rightX)];
+  const bottomIntersection = [solveForX(slope, intercept, bottomY), bottomY];
+  const leftIntersection = [leftX, solveForY(slope, intercept, leftX)];
+  // There will be duplicate intersection points if the line intersects a corner
+  const uniqueIntersections = uniqWith(
+    [topIntersection, rightIntersection, bottomIntersection, leftIntersection],
+    isEqual
+  );
+  return uniqueIntersections
+    .filter(pt => {
+      return pt[0] >= leftX && pt[0] <= rightX && pt[1] >= bottomY && pt[1] <= topY;
```

Comment:
I used `>=` here since I'm calculating the line coordinates at the boundaries, so I always expect one of the numbers to be right on the edge - otherwise, I'd be able to use `board.hasPoint`.

This code assumes that there are always exactly 0 or 2 intersections between a line and the bounding box. Reading through your comment, though, I realized I'm not sure what happens in the case where a line is equal to an axis...

Reply:
Looked into this and I think the code works as is. In the case where the line overlaps a boundary, the intersection points for that side will be undefined and get filtered out here